probe-rs/src/probe/jlink/mod.rs

```diff
@@ -117,45 +125,68 @@ impl ProbeFactory for JLinkFactory {
                 // We detect the proprietary J-Link interface using the vendor-specific class codes
                 // and the endpoint properties
                 if descr.class() == 0xff && descr.subclass() == 0xff && descr.protocol() == 0xff {
-                    if let Some((intf, _, _)) = jlink_intf {
-                        Err(JlinkError::Other(format!(
-                            "found multiple matching USB interfaces ({} and {})",
-                            intf,
-                            descr.interface_number()
-                        )))?;
-                    }
-
                     let endpoints: Vec<_> = descr.endpoints().collect();
                     trace!("endpoint descriptors: {:#x?}", endpoints);
                     if endpoints.len() != 2 {
-                        warn!("vendor-specific interface with {} endpoints, expected 2 (skipping interface)", endpoints.len());
+                        debug!("vendor-specific interface with {} endpoints, expected 2 (skipping interface)", endpoints.len());
                         continue;
                     }
 
                     if !endpoints
                         .iter()
                         .all(|ep| ep.transfer_type() == EndpointType::Bulk)
                     {
-                        warn!(
+                        debug!(
                             "encountered non-bulk endpoints, skipping interface: {:#x?}",
                             endpoints
                         );
                         continue;
                     }
 
+                    let Some(index) = descr.string_index() else {
+                        debug!("interface lacks string descriptor, skipping interface");
+                        continue;
+                    };
+
+                    let Ok(interface_name) =
+                        handle.get_string_descriptor(index, language, Duration::from_millis(100))
+                    else {
+                        debug!(
+                            "failed to read interface name string descriptor, skipping interface"
+                        );
+                        continue;
+                    };
+
+                    if let Some((intf, ref name, _, _)) = jlink_intf {
+                        const EXPECTED_INTERFACE_NAME: &str = "BULK interface";
```

Comment:
> I'm a bit worried that this just breaks for other J-links now

Note that this check is only done if there is a duplicate vendor-specific interface. We previously rejected all those devices anyway. I'm also worried this will bring a world of pain, but a 5+ year old edu mini reports the same name so there's a chance it's at least not changing every day.